SnoopCompileCore/src/snoopc.jl

```diff
@@ -4,33 +4,50 @@ using Serialization
 
 """
 ```
-@snoopc "compiledata.csv" begin
-    # Commands to execute, in a new process
+@snoopc [pspawn=true] [String["<julia_flags...>"]] "compiledata.csv" begin
+    # Commands to execute, either in a new process if pspawn==true, or via eval if false.
 end
 ```
 causes the julia compiler to log all functions compiled in the course
 of executing the commands to the file "compiledata.csv". This file
 can be used for the input to `SnoopCompile.read`.
+
+Julia flags can optionally be passed as a string or an array of strings, which will be set
+on the newly spawned julia process if `pspawn=true`. If `pspawn=false`, setting
+`julia_flags` will be ignored.
+
+If `pspawn=false`, the commands will be run in the same julia process, via `eval()` in
+the current module. This will only report _new_ compilations, that haven't already been
+cached in the current julia process, which can be (carefully) used to prune down the
+results to only the code you are interested in.
 """
-macro snoopc(flags, filename, commands)
-    return :(snoopc($(esc(flags)), $(esc(filename)), $(QuoteNode(commands))))
+macro snoopc(args...)
+    @assert 4 >= length(args) >= 2 """Usage: @snoopl [args...] "snoopl.csv" "snoopl.yaml" commands"""
+    flags, (filename, commands) = args[1:end-2], args[end-1:end]
+    pspawn_expr, julia_flags = begin
+        if length(flags) == 2
+            flags[1], flags[2]
+        elseif flags[1].head == :(=) && flags[1].args[1] == :pspawn
+            flags[1], String[]
+        else
+            :(pspawn=false), flags[1]
+        end
+    end
+    return :(snoopc($(esc(julia_flags)), $(esc(filename)), $(QuoteNode(commands)), $__module__; $(esc(pspawn_expr))))
 end
 macro snoopc(filename, commands)
     return :(snoopc(String[], $(esc(filename)), $(QuoteNode(commands))))
 end
 
-function snoopc(flags, filename, commands)
-    println("Launching new julia process to run commands...")
-    # addprocs will run the unmodified version of julia, so we
-    # launch it as a command.
+function snoopc(flags, filename, commands, _module=nothing; pspawn=true)
+    if pspawn
+        println("Launching new julia process to run commands...")
+    end
     code_object = """
             using Serialization
-            while !eof(stdin)
-                Core.eval(Main, deserialize(stdin))
-            end
+            Core.eval(Main, deserialize(IOBuffer(read(stdin))))
```

Comment:
Any particular reason for this change? What if `stdin` is *huge*? Likewise elsewhere among the changes.

Is it possible that this is the change that causes the closed stream issue? It seems possible that Julia 1.0 is not being as careful as this code.

Reply:
Alas, this is not responsible for the error -- this was my attempt to fix it, but it didn't help. I can happily revert these changes, yeah.